### PR TITLE
Update payment customization templates and samples for 2023-10

### PR DESCRIPTION
**Pull request by nickwesselman**
Update payment customizations for:
* New unified config format
* Use of function targets
* Localization of title and description
* New API version and schemas

This may be easier to visualize in an [example app](https://github.com/Shopify/nw-scratch/tree/new-function-templates/payment-customization-sample/extensions) that I've used to 🎩 these changes.

Corresponding tutorial updates will be made [here](https://github.com/Shopify/shopify-dev/pull/37049) before this is released.

## 🎩 
Ensure your app uses CLI 3.49.4. Creating the sample or an empty app with `@latest` will ensure this, or you can run `pnpm shopify upgrade`.

### The sample
```
pnpm create @shopify/app@latest --template https://github.com/Shopify/function-examples/sample-apps/payment-customizations#payment-customization-2023-10
cd <path-to-app>
pnpm shopify app dev
```

### The templates
```
pnpm shopify app generate extension  --clone-url https://github.com/Shopify/function-examples#payment-customization-2023-10
```

### Rust and JS and Wasm
- [x] `input.graphql.liquid` ➡️  `src\run.graphql.liquid` (just `run.graphql.liquid` for wasm)
- [x] `run.graphql.liquid` `name` ➡️ `handle`
- [x] New `shopify.extension.toml.liquid` with `2023-10`
- [x] Add `locales\en.default.json.liquid`
- [x] Update `schema.graphql`
- [x] Remove `metadata.json` if present

### JavaScript
- [x] GraphQL `Input` ➡️  `RunInput`
- [x] `index.liquid` ➡️  `run.liquid`
- [x] `run.liquid` `InputQuery` ➡️ `RunInput` (js and ts)
- [x] `run.liquid` `FunctionResult` ➡️ `FunctionRunResult` (js and ts)
- [x] `run.liquid` arrow function export ➡️  `export function run(input)` (js and ts)
- [x] `run.liquid` switch to `import type` for `RunInput` and `FunctionRunResult` (ts)
- [x] `index.test.liquid` ➡️ `run.test.liquid`
- [x] `run.test.liquid` change imported function (e.g. `productDiscounts` ➡️ `run`) (js and ts)
- [x] New `index.liquid` with module export
- [x] `package.json.liquid` new `codegen` config

### Rust
- [x] `.gitignore` `.output.graphql` ➡️  `*.output.graphql`
- [x] `Cargo.toml.liquid` `{{name}}` ➡️  `{{handle}}`
- [x] `Cargo.toml.liquid` `shopify_function` to `0.4.0`
- [x] Ensure `shopify.extension.toml.liquid` includes `path` and `watch`
- [x] `shopify.extension.toml.liquid` `{{name}}` ➡️  `{{handle}}`
- [x] `main.rs` ➡️  `run.rs`
- [x] `run.rs` remove `generate_types` and `#[shopify_function]`, use `shopify_function_target`
- [x] `run.rs` `fn function` ➡️  `fn run`
- [x] `run.rs` `FunctionResult` ➡️  `FunctionRunResult`
- [x] Move tests into `run.rs`, delete `tests.rs`
- [x] New `main.rs` with `pub mod run` and placeholder `fn main()`

### CODE REVIEW

File: sample-apps/payment-customizations/extensions/payment-customization-js/src/run.graphql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-query Input {
```
Comment (AsFal) [inline sample-apps/payment-customizations/extensions/payment-customization-js/src/run.graphql:1]: Nit: Is there a reason why this name isn't consistent across javascript and rust?

File: checkout/wasm/payment-customization/default/shopify.extension.toml.liquid [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+  export = "run"
+
+  [extensions.build]
+  command = "echo 'build the wasm'"
+  path = ""
+  watch = []
+
+  [extensions.ui.paths]
+  create = "/"
+  details = "/"
```
Comment (AsFal) [inline checkout/wasm/payment-customization/default/shopify.extension.toml.liquid:21]: Nit: Could you add these new lines at the end of the file?

**Review by AsFal — APPROVED** [review]
Looks good to me!

File: sample-apps/payment-customizations/extensions/payment-customization-js/src/run.graphql (hunk 1 above)
Comment (nickwesselman) [reply sample-apps/payment-customizations/extensions/payment-customization-js/src/run.graphql:1]: Code generation requirements 😿 ... we'd eventually like to align Rust codegen with what we do for JS/TS to resolve this.

**Pull request merged**